src/Hamiltonians/HOCartesianCentralImpurity.jl

```diff
@@ -0,0 +1,200 @@
+"""
+    log_abs_oscillator_zero(n)
+
+Compute the logarithm of the absolute value of the ``n^\\mathrm{th}`` 1D 
+harmonic oscillator function evaluated at the origin. The overall sign is
+determined when the matrix element is evaluated in [`ho_delta_potential`](@ref).
+"""
+function log_abs_oscillator_zero(n)
+    isodd(n) && return -Inf, 1.0
+    x, _ = SpecialFunctions.logabsgamma((1-n)/2)
+    y, _ = SpecialFunctions.logabsgamma(n+1)
+    result = log(pi)/4 + (n/2)*log(2) - x - y/2
+    return result
+end
+
+"""
+    ho_delta_potential(S, i, j; vals)
+
+Returns the product of two harmonic oscillator functions 
+evaluated at the origin, 
+```math
+    v_{ij} = \\phi_\\mathbf{n}_i(0) \\phi_\\mathbf{n}_j(0)
+```
+which is only non-zero for even-parity states.
+Mode indices `i,j` map to ``D``-dimensional tuples of harmonic oscillator 
+indices ``\\mathbf{n}_i``. `S` defines the bounds of the Cartesian HO modes.
+The values ``\\phi_i(0)`` are calculated with [`log_abs_oscillator_zero`](@ref) 
+and should be precomputed by [`HOCartesianCentralImpurity`](@ref) and passed 
+in as the vector `vals`.
+"""
+function ho_delta_potential(S, i, j; 
```

Comment:
Should there be an observable `G2AtOrigin` that makes use of `ho_delta_potential`? Is that coming in another PR?

Reply:
For 2 particles, (i.e. 1 in this hamiltonian) just use the keyword `impurity_only = true` to get the same operator

Reply:
Fair enough. But this should work for any number of particles, no?

Reply:
Yes this will work for any number of particles